### PR TITLE
RC: 0.11.1 

### PR DESCRIPTION
**Pull request by eikonomega**
Allow users to pass a `<span>` element to header components.

### CODE REVIEW

File: src/components/typography/headers/Header.tsx [hunk 1]
```diff
@@ -7,8 +7,8 @@ import useUITheme from '../../theming/useUITheme';
 export type HeaderProps = {
   /** Underlying HTML element tag to use. */
   size: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
-  /** The text of the header. */
-  text: string;
+  /** The text of the header. This must either be a string or a <span> element. */
+  text: string | ReactElement<HTMLSpanElement>;
```
Comment (bobular) [inline src/components/typography/headers/Header.tsx:11]: So the `<HTMLSpanElement>` is a hint for the developer but isn't enforced by TypeScript?  (As you note below.)

File: src/stories/typography/Headers.stories.tsx [hunk 2]
```diff
@@ -42,6 +42,28 @@ Default.args = {
   useTheme: false,
 };
 
+export const FormattedText: Story<Omit<HeaderProps, 'size'>> = (args) => {
```
Comment (bobular) [inline src/stories/typography/Headers.stories.tsx:45]: I think you can re-use the `Default` story 'component' like I did in my draft PR:

https://github.com/VEuPathDB/core-components/pull/39/files#diff-142e234956197d52ff648da0365d00fb979141d334b5593d19f7e660c3c58d48R45-R50

**Review by bobular — APPROVED** [review]
Looks great. Let me know when it's on npm!

File: src/components/typography/headers/Header.tsx (hunk 1 above)
Comment (dmfalke) [reply src/components/typography/headers/Header.tsx:11]: @eikonomega and I spoke briefly on slack about this. I think the plan is to make this:

```suggestion
  text: ReactNode;
```

It's a little more flexible for unforeseen scenarios.

**Review by dmfalke — APPROVED** [review]

**Pull request merged**